packages/zoe/src/contractFacet/contractFacet.js

```diff
@@ -0,0 +1,196 @@
+// @ts-check
+
+// This is the Zoe contract facet. Each time we make a new instance of a
+// contract we will start by creating a new vat and running this code in it. In
+// order to install this code in a vat, Zoe needs to import a bundle containing
+// this code. We will eventually have an automated process, but for now, every
+// time this file is edited, the bundle must be manually rebuilt with
+// `yarn build-zcfBundle`.
+
+import { assert, details } from '@agoric/assert';
+import { E } from '@agoric/eventual-send';
+import makeWeakStore from '@agoric/weak-store';
+
+import { areRightsConserved } from './rightsConservation';
+import { makeIssuerTable } from '../issuerTable';
+import { assertKeywordName, getKeywords } from '../zoeService/cleanProposal';
+import { evalContractBundle } from './evalContractCode';
+import { makeSeatAdmin } from './seat';
+import { makeExitObj } from './exit';
+
+import '../../exported';
+import '../internal-types';
+
+export function buildRootObject() {
+  /** @type ExecuteContract */
+  const executeContract = async (
+    bundle,
+    zoeService,
+    invitationIssuer,
+    zoeInstanceAdmin,
+    hardenedInstanceRecord,
+  ) => {
+    const instanceRecord = { ...hardenedInstanceRecord };
```

Comment:
We're working to hard to update the record in place, by mutating properties (which is outside Jessie anyway), and then doing a bunch of defensive copying and hardening around that to protect ourselves from these mutations. Why not instead just declare this as 
```js
let instanceRecord = harden({...whatever});
```
and then reassign the variable with an updated record on lines 107 and 111. Then, we can avoid the copy-and-harden on line 144.

Reply:
Fixed by #1392